examples/gno.land/r/demo/authbanker/authbanker.gno

```diff
@@ -0,0 +1,78 @@
+package authbanker
+
+import (
+	"errors"
+	"std"
+	"strings"
+
+	"gno.land/p/demo/auth"
+	"gno.land/p/demo/ufmt"
+	"gno.land/r/demo/authreg"
+)
+
+// This example is there mostly to demonstrate auth usage, it is quite limited
+// only EOAs can fund accounts
+
+const denom = "ugnot"
+
+var vaults = make(map[string]int64)
+
+// GetCoins returns the amount of coins in the vault identified by `entityID`
+func GetCoins(entityID string) int64 {
+	return vaults[entityID]
+}
+
+// SendCoins sends `amount` coins from the vault identified by `atok` to the account identified by `to`.
+//
+// `to` can be an entity ID or an address.
+func SendCoins(atok auth.Token, to string, amount int64) {
+	if amount < 1 {
+		panic("sent amount must be >= 0")
+	}
+
+	from := authreg.Authenticate(atok)
```

Comment:
What prevents me from creating my own authenticator? I don't understand where you expect to whitelist the approved authenticators.

Reply:
Nothing, the pattern is meant to be extendable.

The registry namespaces the authenticators so you can't authenticate an entity from an other authenticator.

How would you exploit this?

Reply:
Actually there was a vulnerability in namespacing if you pass paths with `..`, maybe you were refering to that? I added a guard against that [here](https://github.com/gnolang/gno/pull/3406/commits/9a31a60a772822fa6ed1a941e9818a9077f27520)

Reply:
Also if you want to whitelist providers somewhere, you can check the token's `Source` or statically import the providers and call their Authenticate function

Reply:
I was considering creating a new authenticator, registering it, and then using it for universal acceptance.

I see two secure approaches: one is to implement a whitelist system on the registry itself, and the other is to establish a whitelist system within the contract that checks the authentication.

However, allowing anyone to create an authenticator and contracts to simply "verify if a token is valid" is definitely insecure.